### PR TITLE
feat(ctb): MPT Fuzzing contract changes

### PR DESCRIPTION
**Pull request by clabby**
# Overview

Adds a fuzz test input generator for testing Bedrock's `MerkleTrie.sol`.

Go version of #3847

**Tests**

Adds fuzz tests for `MerkleTrie`:
- `testFuzz_get_validProofs_success`
- `testFuzz_get_invalidRoot_reverts`
- `testFuzz_get_extraProofElements_reverts`
- `testFuzz_get_invalidLargeInternalHash_reverts`
- `testFuzz_get_invalidInternalNodeHash_reverts`
- `testFuzz_get_corruptedProof_reverts`
- `testFuzz_get_invalidDataRemainder_reverts`
- `testFuzz_get_prefixedValidKey_reverts`
- `testFuzz_get_emptyKey_reverts`
- `testFuzz_get_partialProof_reverts`

**Metadata**
ENG-3048

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* sc/ctb-mini-trie-refactor
  * **PR #4233** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4233" target="_blank">Graphite</a> 
    * **PR #4234** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4234" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4234?utm_source=stack-comment).

**Review by maurelian — APPROVED** [review]

File: packages/contracts-bedrock/contracts/test/MerkleTrie.t.sol [hunk 1]
```diff
@@ -5,59 +5,56 @@ import { CommonTest } from "./CommonTest.t.sol";
...
         bytes memory val = hex"6176616c32";
         bytes[] memory proof = new bytes[](3);
-        proof[0] = hex"e68416b65793a03101b4447781f1e6c51ce76c709274fc80bd064f3a58ff981b6015348a826386";
-        proof[1] = hex"f84580a0582eed8dd051b823d13f8648cdcd08aa2d8dac239f458863c4620e8c4d605debca83206262856176616c32ca83206363856176616c3380808080808080808080808080";
+        proof[
+            0
+        ] = hex"e68416b65793a03101b4447781f1e6c51ce76c709274fc80bd064f3a58ff981b6015348a826386";
+        proof[
+            1
+        ] = hex"f84580a0582eed8dd051b823d13f8648cdcd08aa2d8dac239f458863c4620e8c4d605debca83206262856176616c32ca83206363856176616c3380808080808080808080808080";
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/MerkleTrie.t.sol:22]: Looks like a lot of formatter changes. Could we move these into another PR below this one?

**Review by maurelian — CHANGES_REQUESTED** [review]

**Comment by maurelian** [comment]
I think we could also build another PR above this one that deletes some TS files right?

File: packages/contracts-bedrock/contracts/test/MerkleTrie.t.sol (hunk 1 above)
Comment (trianglesphere) [reply packages/contracts-bedrock/contracts/test/MerkleTrie.t.sol:22]: ditto

Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/MerkleTrie.t.sol:22]: I put it into #4319 already actually.

**Review by maurelian — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]
Fantastic work here

**Review by tynes — APPROVED** [review]
